samples/sample_script.sh

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env sh
+
+# set -xv
+
+# show usage and quit
+function usage()
+{
+  printf "usage: %s <command>\n" $(basename $0) >&2
+  printf "\n" >&2
+  printf "    <command> is one of: download, remove\n" >&2
+  printf "    <sample_name> is one of: iris, gsod, metriccentury\n" >&2
+  printf "\n" >&2
+  printf "    commands are:\n"
+  printf "        download [<version>] - downloads and mounts all sample data\n" >&2
+  printf "                               gets latest version unless <version> supplied\n" >&2
+  printf "        remove - removes all sample data\n" >&2
+  printf "        version - show local version\n" >&2
+  printf "        versions - list available version\n" >&2
+  exit 2
+}
+
+
+# complain and quit
+function fail_out()
+{
+  printf "Failed! %s\n" "$@" >&2
+  exit 2
+}
+
+
+# check that we have the expected enlistment directory; complain and fail if not
+function require_enlistment()
+{
+  if [ ! -d $target_path/.git ]; then
+     printf "no samples collection at %s; use the download command first\n" $target_path >&2
+     fail_out
+  fi
+}
+
+
+# clone the git repo, don't report progress but DO report errors
+function do_download()
+{
+  git clone --quiet $git_root_url $target_path || fail_out "Couldn't clone samples repository"
+  printf "samples downloaded to $target_path\n"
+  [ ! -z "$1" ] && do_version "$1"
+}
+
+
+# remove the enlistment directory
+function do_remove()
+{
+  [ -d $target_path ] || fail_out "Couldn't find $target_path$root_path"
+  rm -rf $target_path
+  printf "$target_path removed\n"
+}
+
+
+# list all the tags we know
+function do_list_versions()
+{
+  cd $target_path
+  git tag -n
+  printf "Version listed\n"
+}
+
+
+# switch version to something different
+function do_version()
+{
+  cd $target_path
+  git -c advice.detachedHead=false checkout "$1" || fail_out "Couldn't change versions"
+  printf "set to version %s\n" "$1"
+}
+
+
+#####
+# set up the source and target info
+git_root_url="git://github.com/mikeblas/samples-junk.git"
+target_path="/data/samples"
+
+# figure out command and dispatch ...
+case "$1" in
+  download)
+    do_download "$2"
+    ;;
+  version)
+    require_enlistment
+    [ -z "$2" ] && fail_out "need a version specification"
+    do_version "$2"
+    ;;
+  versions)
+    require_enlistment
+    do_list_versions
```

Comment:
1) Should versions return both local and remote versions?  Probably.
2) Similar to the above comment.  Does it make sense for this to fail, tell them what to do, then they do it?  I'm guessing we should either (a) get the list without requiring a download or (b) just do the download first.

Reply:
I think (a) is the better option.  Checking versions shouldn't require a download and a download should be done explicitly by a potential user

Reply:
I think (a) is the better option. Checking versions shouldn't require a download and a download should be done explicitly by a potential user

Reply:
from scratch:

`download` will download, and set to the newest tag
`download v1.1` will download and get tag "v1.1"
`version v1.1` will download without complaint and get tag "v1.1"

with an existing enlistment:
`download` will say "already exists", no action
`download v1.1` will get tag "v1.1"
`version v1.1` will get tag "v1.1"

Reply:
We did #2 and (a)